data-prepper-core/src/main/java/org/opensearch/dataprepper/DataPrepper.java

```diff
@@ -71,6 +75,28 @@ public DataPrepper(
      */
     public boolean execute() {
         peerForwarderServer.start();
+        Set<String> pipelineNames = transformationPipelines.keySet();
+        int numRetries = 0;
+        while (pipelineNames.size() > 0 && numRetries++ < MAX_RETRIES) {
+            Set<String> failedPipelines = new HashSet<String>();
+            Iterator pipelineIter = pipelineNames.iterator();
+            while (pipelineIter.hasNext()) {
+                String pipelineName = (String)pipelineIter.next();
+                if (!transformationPipelines.get(pipelineName).isReady()) {
+                    failedPipelines.add(pipelineName);
+                }
+            }
+            pipelineNames = failedPipelines;
+            if (pipelineNames.size() > 0) {
+                try {
+                    Thread.sleep(5000);
```

Comment:
This could be configurable in data-prepper-config.yaml. But, that could come as a later improvement.